lite2/store/db/db.go

```diff
@@ -207,14 +207,14 @@ func (s *dbs) FirstSignedHeaderHeight() (int64, error) {
 // height. It returns ErrSignedHeaderNotFound if no such header exists.
 //
 // Safe for concurrent use by multiple goroutines.
-func (s *dbs) SignedHeaderAfter(height int64) (*types.SignedHeader, error) {
+func (s *dbs) SignedHeaderBefore(height int64) (*types.SignedHeader, error) {
 	if height <= 0 {
 		panic("negative or zero height")
 	}
 
-	itr, err := s.db.Iterator(
-		s.shKey(height+1),
-		append(s.shKey(1<<63-1), byte(0x00)),
+	itr, err := s.db.ReverseIterator(
+		nil,
+		s.shKey(height),
```

Comment:
shouldn't we start with height - 1?

Reply:
I'm not too sure why because I am not that familiar with the db stuff but when I tried with height -1 it failed the test, so I guess the reverse iterator upper bound does not actually include it

Reply:
Correct. The bounds are `[start, end)`, i.e. start is inclusive and end is exclusive. This is regardless of the iterator direction. Perhaps not very intuitive, but that's how it is today.